avocado/utils/linux_modules.py

```diff
@@ -96,7 +101,23 @@ def loaded_module_info(module_name):
     :rtype: dict
     """
     l_raw = process.system_output('/sbin/lsmod')
-    return parse_lsmod_for_module(l_raw, module_name)
+    modinfo_dic = parse_lsmod_for_module(l_raw, module_name)
+    output = process.system_output("/sbin/modinfo %s" % module_name)
+    if output:
+        param_list = []
+        for line in output.splitlines():
+            key = line.split()[0]
+            if key == 'filename:':
+                modinfo_dic['filename'] = line.split()[-1]
```

Comment:
It's not split many times, it's split in individual if/elif branches and the split is not needed when the key is unknown